node/core/pvf/worker/src/prepare.rs

```diff
@@ -109,8 +111,22 @@ pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 			// Spawn another thread for preparation.
 			let prepare_fut = rt_handle
 				.spawn_blocking(move || {
+					#[cfg(feature = "wrapper-allocator")]
+					ALLOCATOR_DATA.checkpoint();
```

Comment:
A few questions:
1. This will measure only the dynamically allocated memory,  it will fail to monitor any unexpected increase in the stack, is that what we want ?
2. How is this different from the bellow `get_max_rss_thread` and why isn't that enough ?

Reply:
> 2. How is this different from the bellow get_max_rss_thread and why isn't that enough ?

The issue with that is it is not deterministic enough. (Is mainly being used for gathering metrics right now.) Different kernel configurations/versions may manage memory differently (simple example is some validators may have swap enabled and some not). So they may get different values for the resident memory (how much is actually held in RAM). So some validators may reach the limit and others not.

Reply:
> 1. This will measure only the dynamically allocated memory,  it will fail to monitor any unexpected increase in the stack, is that what we want ?

I hope yes, in the preparation phase we're not executing any untrusted code so we just presume that Wasmtime guys know what they're doing and won't abuse the stack usage. We only bother about malicious Wasm code that could force the compiler to allocate a lot of memory.

> 2. How is this different from the bellow `get_max_rss_thread` and why isn't that enough ?

A lot of concerns here.
* Not supported on every platform
* Allocated memory is not necessarily resident
* Wasmtime may spawn its own threads and we'd lose their allocations in the per-thread accounting